src/types/subscriptions.ts

```diff
@@ -0,0 +1,34 @@
+import type { DwnError } from '../core/dwn-error.js';
+import type { GenericMessageReply } from '../types/message-types.js';
+import type { MessageStore } from './message-store.js';
+import type { EventsSubscribeMessage, EventsSubscription } from './events-types.js';
+import type { Filter, KeyValues } from './query-types.js';
+import type { GenericMessage, GenericMessageHandler, GenericMessageSubscription } from './message-types.js';
+
+export type EmitFunction = (tenant: string, message: GenericMessage, indexes: KeyValues) => void;
+
+/**
+ * The EventStream interface implements a pub/sub system based on Message filters.
+ */
+export interface EventStream {
+  subscribe(tenant: string, message: EventsSubscribeMessage, filters: Filter[], messageStore: MessageStore): Promise<EventsSubscription>;
+  subscribe(tenant: string, message: GenericMessage, filters: Filter[], messageStore: MessageStore): Promise<GenericMessageSubscription>;
```

Comment:
For my understanding: why would we need the `GenericMessage` version of the `subscribe()` method?

Reply:
I can remove it from this PR, but this is because I was overloading the method with a subscription signature for `RecordsSubscribeMessage`.

In the original PR I made a comment that I am 50/50 on the abstraction and overloading vs just having `GenericMessage` and a generic handler. I see pluses for either argument. Would be happy to chat about that if you'd like.

The only current difference between the two returned objects is that the handler for `RecordsSubscription` will emit `RecordsWriteMessage` and `RecordsDeleteMessage` only vs `GenericMessage`.